transaction.go

```diff
@@ -49,198 +90,303 @@ type Transaction struct {
 
 	// The wal that was used to create the transaction
 	wal *WAL
```

Comment:
This comment applies to code that's not in the diff, but I don't think  we need the `finalPage` field in the transaction anymore - we put the checksum into the first page, which was the only thing we needed from the final page iirc.